helix-term/src/ui/markdown.rs

```diff
@@ -215,10 +215,32 @@ impl Component for Markdown {
     }
 
     fn required_size(&mut self, viewport: (u16, u16)) -> Option<(u16, u16)> {
-        let contents = parse(&self.contents, None, &self.config_loader);
         let padding = 2;
-        let width = std::cmp::min(contents.width() as u16 + padding, viewport.0);
-        let height = std::cmp::min(contents.height() as u16 + padding, viewport.1);
+        if padding >= viewport.1 || padding >= viewport.0 {
+            return None;
+        }
+        let contents = parse(&self.contents, None, &self.config_loader);
+        let max_text_width = viewport.0 - padding;
```

Comment:
We should limit the width to max 120 chars, otherwise the text gets too hard to read.

`let max_text_width = (viewport.0 - padding).min(120 + padding);`

Reply:
You actually don't need to calculate `max_text_width` because you can use `contents.width()`.

`let max_width = contents.width().min(120 + padding)`

Reply:
I had limited max width with 120.
How to calculate content's height if we don't calcutlate `max_text_width`?